src/main/java/org/broadinstitute/hellbender/tools/walkers/cancer/contamination/ContaminationResults.java

```diff
@@ -0,0 +1,248 @@
+
+                // if it exists... just merge it
+                if (!storedData.get(aggregationKey).containsKey(pop)) {
+                    storedData.get(aggregationKey).put(pop, new ArrayList<>());
+                }
+
+                double[] newData = new double[newStats.getContamination().getBins().length];
+                System.arraycopy(newStats.getContamination().getBins(),0,newData,0,newStats.getContamination().getBins().length);
+                storedData.get(aggregationKey).get(pop).add(new ContaminationData(newStats.getSite(), newStats.getBasesMatching(), newStats.getBasesMismatching(), newData));
+            }
+
+
+
+            // merge the sets
+            if (stats.containsKey(aggregationKey)) {
+
+                // and for each population
+                for (String pop : populationContaminationStats.keySet()) {
+                    ContaminationStats newStats = populationContaminationStats.get(pop);
+
+                    // if it exists... just merge it
+                    if (stats.get(aggregationKey).containsKey(pop)) {
+                        stats.get(aggregationKey).get(pop).add(newStats);
+                    } else {
+                        stats.get(aggregationKey).put(pop, newStats);
+                    }
+                }
+            } else {
+                stats.put(aggregationKey, populationContaminationStats);
+            }
+        }
+    }
+
+    /**
+     * output the contamination data, and return the contamination data
+     * @param out the output source
+     * @return the contamination value
+     */
+    public void outputReport(double precision, PrintStream out, double fractionToTrim, double trimInterval, double betaThreshold) {
+        out.println("name\tpopulation\tpopulation_fit\tcontamination\tconfidence_interval_95_width\tconfidence_interval_95_low\tconfidence_interval_95_high\tsites");
+
+        for (Map.Entry<String,Map<String, ContaminationStats>> entry : stats.entrySet()) {
+            for (ContaminationStats stats : entry.getValue().values()) {
+                String aggregationLevel = entry.getKey();
+                String population = stats.getContamination().getPopultationName();
+
+                List<ContaminationData> newStats = storedData.get(aggregationLevel).get(population);
+                String pm = "%3." + Math.round(Math.log10(1/precision)) +"f";
+
+                int bins = newStats.iterator().next().getBins().length;
+                int maxTrim = (int) Math.floor((double)(newStats.size()) * fractionToTrim);
+
+                // sort the collection
+                Collections.sort(newStats);
+
+                List<ContaminationData> data = new ArrayList<ContaminationData>(newStats);
+
+                // trim sites with > 95% p of being > 0.5 f (based on beta distribution)
+                int trimmed = 0;
+                for(Iterator<ContaminationData> i = data.iterator(); trimmed < maxTrim && i.hasNext();) {
```

Comment:
Instead of removing from an iterator, then doing a loop, just have a filtered stream with a foreach, if possible